### PR TITLE
Update user guide

### PR DESCRIPTION
**Pull request by brelkh**
Add information for `edit` and `clear` command.

Edit some other lines for consistency and clarity.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -147,7 +147,33 @@ Format : `unmark i/STUDENT_ID DONE_TASK_INDEX`
 Example:
 * `unmark i/AXXXXXXXR 1`
     * Marks the first done task for the student with student ID AXXXXXXXR as undone.
-    
+
+### Editing a student's information: `edit`
+
+Edits a student's information in TAPA.
+
+Format: `edit i/STUDENT_ID n/STUDENT_NAME m/MODULE_CODE [p/PHONE_NUMBER] [h/TELEGRAM_HANDLE] [e/EMAIL_ADDRESS]​`
```
Comment (Benjykoh) [inline docs/UserGuide.md]:
```suggestion
Format: STUDENT_INDEX `edit [i/STUDENT_ID] [n/STUDENT_NAME] [m/MODULE_CODE] [p/PHONE_NUMBER] [h/TELEGRAM_HANDLE] [e/EMAIL_ADDRESS]​`
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -147,7 +147,33 @@ Format : `unmark i/STUDENT_ID DONE_TASK_INDEX`
...
+
+* The index of the student to be edited is a compulsory field.
+* The student’s matriculation number, name, module code, phone number, telegram handle, and email address fields are optional and can be excluded.
+
```
Comment (Benjykoh) [inline docs/UserGuide.md:159]: Perhaps we could add what happens if invalid index is given (similar to the commands like "mark" and "unmark")

**Comment by Benjykoh** [comment]
Thanks for the additional edit! Almost LGTM, but I think the formatting for the "edit" command has some inconsistencies (The description and format does not use an index to identify which student to edit, while the example does)

**Comment by brelkh** [comment]
> 

@Benjykoh  Thanks for pointing that out! Edited it with this change

File: docs/UserGuide.md (hunk 2 above)
Comment (brelkh) [reply docs/UserGuide.md:160]: Right, I'll get that done then!

Comment (brelkh) [reply docs/UserGuide.md:160]: @Benjykoh I'll resolve this for now, thanks and let me know if you see any other improvements though :)

File: docs/UserGuide.md [hunk 3]
```diff
@@ -147,7 +147,34 @@ Format : `unmark i/STUDENT_ID DONE_TASK_INDEX`
...
+Edits a student's information in TAPA.
+
+Format: `edit STUDENT_INDEX [i/STUDENT_ID] [n/STUDENT_NAME] [m/MODULE_CODE] [p/PHONE_NUMBER] [h/TELEGRAM_HANDLE] [e/EMAIL_ADDRESS]​`
+
+* The index of the student to be edited is a compulsory field.
+* The student’s matriculation number, name, module code, phone number, telegram handle, and email address fields are optional and can be excluded.
+* An error message will be displayed to the user if the specified index is a negative number or larger than the number of students in TAPA.
+
+Example:
+* `edit 10 m/CS2103T p/98765432 t/@john e/E0123456Z`
```
Comment (Korochii) [inline docs/UserGuide.md:162]:
```suggestion
* `edit 10 m/CS2103T p/98765432 t/johnnn e/e0123456@u.nus.edu`
```
I realised there were typos in our original draft user guide for the format of edit and add... (will change the one for add later)

**Review by Korochii — COMMENTED** [review]
Small typo which should be easy to fix

File: docs/UserGuide.md (hunk 3 above)
Comment (brelkh) [reply docs/UserGuide.md:162]: ohh, I see, I wasn't too sure which email format to use in the user guide but I'll commit your suggestions first! 👍

**Review by Benjykoh — APPROVED** [review]

**Comment by Benjykoh** [comment]
Alright LGTM :)

**Pull request merged**